App.js

```diff
@@ -1,129 +1,148 @@
-This is a list item 1
-</li>
-<li data-wp-block-name='core/list-item'>
-This is a list item 2
-</li>
-</ul>`;
+globalThis.__10up__HEADLESS_CONFIG = { ...headlessConfig };
 
+import { StatusBar } from "expo-status-bar";
+import { FlatList, Image, StyleSheet, Text, View } from "react-native";
+// metro doesn't support package.json exports field @10up/headless-core/react works with web bundlers.
+import { BlocksRenderer } from "@10up/headless-core/dist/cjs/react/components/BlocksRenderer";
+import {
+  HeadingBlock,
+  ImageBlock,
+  ParagraphBlock,
+  SettingsProvider,
+  useFetchPost,
+} from "@10up/headless-core/dist/cjs/react";
+import { isBlockByName } from "@10up/headless-core";
 
 /**
  * Trims raw text nodes
- * 
- * @param {*} param0 
- * @returns 
+ *
+ * @param {*} param0
+ * @returns
  */
 const RawText = ({ domNode }) => {
   return domNode.data.trim();
-}
+};
 
 RawText.defaultProps = {
   /**
    * Catches any non-orphans and non-empty text fields
-   * 
-   * @param {*} node 
-   * @returns 
+   *
+   * @param {*} node
+   * @returns
    */
-  test: (node) => node.type === 'text' && node.parent !== null && node.data?.trim().length >= 0
-}
-
-const ParagraphBlock = ({ children }) => {
-  return <Text>[Text Component]{children}[Text Component]</Text>
+  test: (node) =>
+    node.type === "text" &&
+    node.parent !== null &&
+    node.data?.trim().length >= 0,
 };
 
-const ListWrapperBlock = ({ children }) => {
-  return <View><Text>[React Native List Component]</Text>{children}<Text>[React Native List Component]</Text></View>
-}
-
-const ListItemWrapperBlock = ({ children }) => {
-  return <Text>[React Native List Item]{children}[React Native List Item]</Text>
-}
-
-/**
- * In this example FlatListBlock goes over its children to build an actual list.
- * 
- * This assumes this the list is text only
- * 
- * @param {*} param0 
- * @returns 
- */
-const FlatListBlock = ({ children }) => {
-  // note: at this point children has already recursively gone through the "dom-to-react" conversion
-  const items = children.filter((node) => node.type === 'li').map(child => { return child.props.children; });
-  
-  const renderItem = ({ item }) => {
-    return (
-      <Text>{item}</Text>
-    );
-  }
-  return <FlatList data={items} renderItem={renderItem} />
-}
-
 /**
  * A noop block
- * 
+ *
  * You can also use this to create a "allowList" of blocks and ignore blocks you don't want to render
- * 
- * @returns 
+ *
+ * @returns
  */
-const EmptyBlock = () => <></>
+const EmptyBlock = () => <></>;
 
 EmptyBlock.defaultProps = {
   /**
    * Essentially catches any empty text nodes to make sure they don't get rendered under a <View> component
-   * 
-   * @param {*} node 
-   * @returns 
+   *
+   * @param {*} node
+   * @returns
    */
   test: (node) => {
-    return node.type === 'text' && (node.parent === null || node.data?.trim().length === 0);
+    const allowedBlocks = ["core/paragraph", "core/heading", "core/image"];
```

Comment:
In the empty block I'm skipping anything that's not in the allowed block list, this allow me to only render the stuff I want.